app/src/js/components/BatchAsyncCommands/BatchAsyncCommands.js

```diff
@@ -32,201 +32,185 @@ const CONCURRENCY = 3;
+  const confirmAction = () => {
     if (typeof getModalOptions === 'function') {
       const modalOptions = getModalOptions({
         selected,
         history,
         isOnModalConfirm: true,
         isOnModalComplete: false,
-        closeModal: this.closeModal,
+        closeModal
       });
-      this.setState({ modalOptions });
+      setModalOpts(modalOptions);
 
       // if we're replacing the onConfirm function, we don't want to continue with the current one
       if (modalOptions.onConfirm) {
         return;
       }
     }
-    if (!this.isInflight()) this.start();
-  }
+    if (!isInflight()) start();
+  };
 
-  cancel() {
+  const cancel = () => {
     // prevent cancel when inflight
-    if (!this.isInflight()) this.setState({ activeModal: false });
-  }
+    if (!isInflight()) setActiveModal(false);
+  };
 
-  start() {
-    const { selected } = this.props;
+  const start = () => {
     // if we have inflight callbacks, don't allow further clicks
-    if (!Array.isArray(selected) || !selected.length || this.isInflight()) { return false; }
-    const q = queue(CONCURRENCY);
-    for (let i = 0; i < selected.length; i += 1) {
-      q.add(this.initAction, this.buildId(selected[i]));
-    }
-    q.done(this.onComplete);
-  }
+    if (!Array.isArray(selected) || !selected.length || isInflight()) { return false; }
+    const q = queue(initAction, CONCURRENCY);
+    selected.forEach((item) => {
+      q.add(initAction, buildId(item));
+    });
+    q.done(onComplete);
+  };
 
   // save a reference to the callback in state, then init the action
-  initAction(id, callback) {
-    const { dispatch, action } = this.props;
-    const { callbacks, meta } = this.state;
-    callbacks[id] = callback;
-    this.setState({ callbacks });
+  const initAction = (id, callback) => {
+    setCallbacks((prevCallbacks) => ({
+      ...prevCallbacks,
+      [id]: callback
+    }));
     return dispatch(action(id, meta));
-  }
+  };
 
   // immediately change the UI to show either success or error
-  onComplete(errors, results) {
-    const {
-      getModalOptions,
-      selected,
-      history,
-    } = this.props;
+  const onComplete = (errors) => {
     // turn array of errors from queue into single error for ui
-    const errorMessage = this.createErrorMessage(errors);
-    this.setState({ errorMessage, results, status: errorMessage ? 'error' : 'success' });
-    if (typeof getModalOptions === 'function') {
-      // setTimeout(() => {
-      const modalOptions = getModalOptions({
-        history,
-        selected,
-        results,
-        errors,
-        errorMessage,
-        isOnModalComplete: true,
-        closeModal: this.closeModal,
-      });
-      this.setState({ modalOptions, status: null });
-    }
-  }
+    const errMsg = createErrorMessage(errors);
+    setErrorMsg(errMsg);
+    setStatus(errMsg ? 'error' : 'success');
+    setResults(errors ? [] : selected.map((item) => buildId(item)));
+    setTimeout(() => {
```

Comment:
In general it delays/schedules the execution of a function- In this case it's set to 0 so that other tasks can be executed asynchronously. There is no wait time before the code is executed.